Delete session shion

nowknowing opened the pull request:
Delete SessionException. Change corresponding tests.
Add SessionBuilder.
Edit TypicalStudent to add one session each.

Add DeleteSession and DeleteStudent.

nowknowing committed:
Add deletesession parser, test class. Comand execution undone.

nowknowing committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2103T-T11-1/tp into delete-session-shion

# Conflicts:
#	src/main/java/seedu/address/logic/parser/CliSyntax.java

nowknowing committed:
Delete SessionException, change tests.
Add DeleteSessionCommand and tests.

samleewy commented on src/test/java/seedu/address/model/session/SessionDateTest.java:
```diff
@@ -24,55 +23,54 @@ public void isValidSessionDate() {
+        assertFalse(SessionDate.isValidSessionDate("2021-13-01", correctTimeValue));
+        assertFalse(SessionDate.isValidSessionDate("2021-00-01", correctTimeValue));
 
         // checks for invalid day
```
Perhaps we could include an assertion for invalid days in month, for example 31 Apr is _invalid_ since April only has 30 days, something similar to leap year check! (does the exception catch this actually?)

JonahhGohh replied:
Good suggestion! The isValidSessionDate() should assertFalse correctly to this test case.

samleewy commented on src/main/java/seedu/address/logic/commands/DeleteSessionCommand.java:
```diff
@@ -0,0 +1,56 @@
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the session identified by the index number used in the student's list of sessions.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " John Doe" + " 1";
```
This should evaluate to `delete_session John Doe 1` but should be `delete_session n/John Doe i/1`

samleewy reviewed (commented):
Minor comments regarding test case and command help text!

nowknowing committed:
parseError

nowknowing committed:
Editting

nowknowing committed:
Fix bug

nowknowing committed:
Fix Bug. Session.toString still unfixed

nowknowing committed:
Bugs fixed. except checkstyle

JonahhGohh commented on src/main/java/seedu/address/model/session/SessionDate.java:
```diff
@@ -23,15 +24,15 @@
-    public SessionDate(String dateValue, String timeValue) throws SessionException {
+    public SessionDate(String dateValue, String timeValue) {
         try {
             LocalDate localDate = LocalDate.parse(dateValue);
             LocalTime localTime = LocalTime.parse(timeValue);
 
             LocalDateTime localDateTime = localDate.atTime(localTime);
             this.dateTime = localDateTime;
         } catch (DateTimeParseException e) {
-            throw new SessionException(INCORRECT_DATE_TIME_FORMAT_ERROR_MESSAGE + e, e);
+            checkArgument(false, MESSAGE_CONSTRAINTS + e.getMessage());
         }
```
Since we have the isValidSessionDate() check now, we do not need this try and catch block anymore since it is a guarantee that we will receive a valid dateValue and timeValue in our SessionDate constructor. 
There are a few changes to other files regarding this issue as well. I will do the changes once this PR is merged.

nowknowing committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2103T-T11-1/tp into delete-session-shion

# Conflicts:
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java

nowknowing committed:
Fix CI bugs

nowknowing committed:
Debug

nowknowing committed:
Debug Checkstyle

nowknowing committed:
CHeckstyle

nowknowing committed:
Checkstyle

nowknowing committed:
Failing test. cannot get studentWithSession

nowknowing committed:
Commented out test as studentWithSession does not have session

nowknowing committed:
Remove session in typical student

nowknowing committed:
fixed Michigan typo in json

JonahhGohh reviewed (commented):
LGTM! 
Are those code that are commented out in DeleteSessionCommandTest and TypicalStudents because they fail the storage test cases?

samleewy reviewed (changes requested):
1. There seem to be missing the switch case for `delete_session` in `AddressBookParser.java`, which makes the command inexecutable.
2. I added the switch case myself locally and tried to execute the command when the student has no sessions, seems to throw an exception, `IndexOutOfBoundsException`, which was out of the application. Maybe you could test it and let me know if you face the same issue too?

Other than the above, the rest looks good! thanks!

nowknowing committed:
Add switch. Debug zeroIndex to OneIndex

nowknowing committed:
wildcard inport fix

nowknowing committed:
Order fix. imports.

nowknowing commented:
> 1. There seem to be missing the switch case for `delete_session` in `AddressBookParser.java`, which makes the command inexecutable.
> 2. I added the switch case myself locally and tried to execute the command when the student has no sessions, seems to throw an exception, `IndexOutOfBoundsException`, which was out of the application. Maybe you could test it and let me know if you face the same issue too?
> 
> Other than the above, the rest looks good! thanks!

I think I've fixed it. 
Still have issues with building TypicalStudent with StudentBuilder.addSession(). Which is why testCases are commented out.But this at the moment is another matter.

nowknowing committed:
EditStudentBuikder to include sessions. Uncomment session added to Typical Student

nowknowing committed:
Student: addSessions method added
StudentBuilder add javadoc

nowknowing committed:
Comment out 2/3 DeleteSessionCommand test.

nowknowing committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2103T-T11-1/tp into delete-session-shion

chwoozy commented:
Pulled to #73
